projects/capnproto/patch.diff

```diff
@@ -0,0 +1,32 @@
+diff --git a/c++/src/capnp/CMakeLists.txt b/c++/src/capnp/CMakeLists.txt
+index c41ff89d..22066fe0 100644
+--- a/c++/src/capnp/CMakeLists.txt
++++ b/c++/src/capnp/CMakeLists.txt
+@@ -291,3 +291,11 @@ if(BUILD_TESTING)
+     add_test(NAME capnp-evolution-tests-run COMMAND capnp-evolution-tests)
+   endif()  # NOT CAPNP_LITE
+ endif()  # BUILD_TESTING
++
++if(DEFINED ENV{LIB_FUZZING_ENGINE})
++  add_executable(fuzz_convert fuzz-convert.c++)
++  target_link_libraries(fuzz_convert capnp kj $ENV{LIB_FUZZING_ENGINE})
++
++  add_executable(fuzz_run fuzz-run.c++ test-util.c++ test-util.h ${test_capnp_cpp_files} ${test_capnp_h_files})
++  target_link_libraries(fuzz_run capnp-rpc capnp kj kj-async capnp-json $ENV{LIB_FUZZING_ENGINE})
++endif()
+diff --git a/c++/src/kj/exception.c++ b/c++/src/kj/exception.c++
+index c8d571ed..fa94f82c 100644
+--- a/c++/src/kj/exception.c++
++++ b/c++/src/kj/exception.c++
+@@ -950,9 +950,11 @@ KJ_THREADLOCAL_PTR(ExceptionCallback) threadLocalCallback = nullptr;
+ 
+ ExceptionCallback::ExceptionCallback(): next(getExceptionCallback()) {
+   char stackVar;
++#ifndef FUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION
```

Comment:
If you remove `ParseErrorCallback`, is this still needed?

Reply:
This one is interesting.
I do not need it when I run the fuzz target on the machine I built it on.
But oss-fuzz uses 2 distinct containers for building and running.
And I got this error running the binary on the base-runner (while run-in this binary on the builder gave no such error)
Should we ask to someone from ossfuzz about it ?

Reply:
If you do see it at all in practice, even after removing the use of `ParseErrorCallback`, then we should keep the `#ifndef`.

Reply:
ok thanks :-)